Bottom: a10d4138ba6ba98d1218678fd1f21a8f6a45f7eb
Top:    07ccf4ee25d2f6927c1bde0ad8beb85ca59f99a2
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-15 10:31:42 -0500

Report build status to Gitlab


---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 518403d2..96df8387 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -12,6 +12,7 @@ buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/cre
 buildsystem_ssh_root_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/root"                , length=20) }}'
 gitlab_buildbot_auth_id:             '{{ lookup("password", playbook_dir + "/credentials/git/auth/oauth_client_id") }}'
 gitlab_buildbot_auth_secret:         '{{ lookup("password", playbook_dir + "/credentials/git/auth/oauth_secret") }}'
+gitlab_buildbot_token:               '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_token") }}'
 gitlab_db_password:                  '{{ lookup("password", playbook_dir + "/credentials/git/db/user"                         , length=20) }}'
 gitlab_email_password:               '{{ lookup("password", playbook_dir + "/credentials/git/email/gitlab@novatech-llc.com"   , length=20) }}'
 gitlab_root_password:                '{{ lookup("password", playbook_dir + "/credentials/git/users/root"                      , length=20) }}'
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index d9b8f54b..72733708 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -20,7 +20,8 @@ RUN pip --no-cache-dir install \
     'buildbot-waterfall-view' \
     'buildbot-www' \
     'ldap3' \
-    'pyOpenSSL'
+    'pyOpenSSL' \
+    'txrequests'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index a50ccfb9..7f2ff830 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -61,6 +61,15 @@ mn = reporters.MailNotifier(
     )
 c['services'].append(mn)
 
+gl = reporters.GitLabStatusPush(
+    os.getenv('GITLAB_TOKEN'),
+    context="Buildbot CI",
+    baseURL=os.getenv("GITLAB_URI"),
+    verbose=True,
+    debug=True
+)
+c['services'].append(gl)
+
 ####### WEB SERVER
 userInfoProvider = util.LdapUserInfo(
         uri=os.getenv('LDAP_URI'),
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index 334a12ce..d261448d 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -51,6 +51,7 @@ container_env:
   GITLAB_URI: 'https://{{ gitlab_hostname }}/'
   GITLAB_OAUTH_ID: '{{ gitlab_buildbot_auth_id }}'
   GITLAB_OAUTH_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_TOKEN: '{{ gitlab_buildbot_token }}'
   GITLAB_WEBHOOK_SECRET: '{{ buildbot_webhook_secret }}'
   LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
   LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
